app/run-test.sh

```diff
@@ -42,13 +41,12 @@ build_cmd="west build ${ZMK_SRC_DIR:+-s $ZMK_SRC_DIR} -d ${ZMK_BUILD_DIR}/tests/
     -b native_posix_64 -p -- -DCONFIG_ASSERT=y -DZMK_CONFIG="$(realpath $path)" \
     ${ZMK_EXTRA_MODULES:+-DZMK_EXTRA_MODULES="$(realpath ${ZMK_EXTRA_MODULES})"}"
 
-if [ -z ${ZMK_TESTS_VERBOSE} ]; then
-    $build_cmd >/dev/null 2>&1
-else
-    echo "ZMK_SRC_DIR:   ${ZMK_SRC_DIR:-.}"
-    echo "ZMK_BUILD_DIR: $ZMK_BUILD_DIR"
-    $build_cmd
-fi
+build_log_tmp="${ZMK_BUILD_DIR}/tmp/$testcase/build.log"
+build_log="${ZMK_BUILD_DIR}/tests/$testcase/build.log"
+mkdir -p $(dirname $build_log_tmp)
+$build_cmd >"$build_log_tmp" 2>&1
+mv "$build_log_tmp" "$build_log"
+rmdir -p $(dirname $build_log_tmp) 2>/dev/null || true
 
 if [ $? -gt 0 ]; then
     echo "FAILED: $testcase did not build" | tee -a ${ZMK_BUILD_DIR}/tests/pass-fail.log
```

Comment:
It would be nice to point callers to the location of the build log when this happens, so they can easily dig into what happened, e.g.:

```suggestion
    echo "FAILED: $testcase did not build (see ${build_log})" | tee -a ${ZMK_BUILD_DIR}/tests/pass-fail.log
```